scripts/pylib/twister/twisterlib/size_calc.py

```diff
@@ -224,3 +259,169 @@ def _calculate_sizes(self):
             self.sections.append({"name": name, "load_addr": load_addr,
                                   "size": size, "virt_addr": virt_addr,
                                   "type": stype, "recognized": recognized})
+
+    def _analyze_elf_file(self) -> None:
+        self._check_elf_file()
+        self._check_is_xip()
+        self._get_info_elf_sections()
+
+    def _get_buildlog_file_content(self) -> typing.List[str]:
+        """Get content of the build.log file.
+
+        @return Content of the build.log file (list[str])
+        """
+        if os.path.exists(path=self.buildlog_filename) and self.buildlog_filename.endswith("build.log"):
+            with open(file=self.buildlog_filename, mode='r') as file:
+                file_content = file.readlines()
+        else:
+            logger.error("Incorrect build.log file to analyze footprints.")
+            file_content = []
+        return file_content
+
+    def _find_offset_of_last_pattern_occurrence(self, file_content: typing.List[str]) -> int:
+        """Find the offset from which the information about the memory footprint is read.
+
+        @param file_content (list[str]) Content of build.log.
+        @return: Offset with information about the memory footprint (int)
+        """
+        result = -1
+        if len(file_content) == 0:
+            logger.warning("Build.log file is empty.")
+        else:
+            # Check the file in reverse order.
+            for idx, line in enumerate(reversed(file_content)):
+                # Condition is fulfilled if the pattern matches with the start of the line.
+                if re.match(pattern=self.PATTERN_SEARCHED_LINE, string=line):
+                    result = idx + 1
+                    break
+        # If the file does not contain information about memory footprint, the warning is raised.
+        if result == -1:
+            logger.warning(msg="Information about memory footprint for this test configuration is not found.")
```

Comment:
this will be a lot of warnings, all qemu tragets will have this issue and the warning does not say anything about where the issue is anyways, I would turn this into debug and add more information about where the issue is.

```
WARNING - Missing information about the memory footprint. Please check build.log.
WARNING - Missing information about memory footprints.
WARNING - Missing information about the memory footprint. Please check build.log.
WARNING - Missing information about memory footprints.
WARNING - Missing information about the memory footprint. Please check build.log.
WARNING - Missing information about memory footprints.
WARNING - Missing information about the memory footprint. Please check build.log.
WARNING - Missing information about memory footprints.
WARNING - Missing information about the memory footprint. Please check build.log.
WARNING - Missing information about memory footprints.
WARNING - Missing information about the memory footprint. Please check build.log.
WARNING - Missing information about memory footprints.
WARNING - Missing information about the memory footprint. Please check build.log.
WARNING - Missing information about memory footprints.
````

Reply:
If it is produced for QEMU I'd say it shouldn't be even with debug verbosity.

Reply:
Comments change to clarify the source of the problem.

Reply:
There are several versions of QEMU-type platforms that differ in architecture. Depending on which platform you choose, you either do not get information about the FLASH memory used, or you get information about the SRAM memory (instead of RAM). This information may not be valuable. 
For this reason, memory footprints are not taken for QEMUs and warnings are not displayed for them.